micro-rdk/src/common/data_collector.rs

```diff
@@ -230,6 +230,14 @@ impl DataCollector {
             data: Some(data),
         })
     }
+
+    pub fn resource_method_key(&self) -> ResourceMethodKey {
```

Comment:
Should this just be `From<DataCollector>` for `ResourceMethodKey`?

Reply:
I considered that, but this would consume the data collector. I forgot that you could do `From<&DataCollector>` though. Would this be preferred?

Reply:
No preference, I'm still learning what is idiomatic for rust and was curious about this. Reading the third bullet of https://doc.rust-lang.org/std/convert/trait.From.html#when-to-implement-from makes me think it should not be done.